packages/model/src/data-providers.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  CalendarCreateDataObject,
+  CalendarDataObject as DefaultCalendarDataObject,
+  CalendarUpdateDataObject,
+  EventCreateDataObject as DefaultEventCreateDataObject,
+  EventDataObject as DefaultEventDataObject,
+  EventUpdateDataObject as DefaultEventUpdateDataObject,
+  FetchEventsOptions,
+  ListCalendarDataObject as DefaultListCalendarDataObject,
+  ListEventDataObject as DefaultListEventDataObject,
+} from '.';
+
+export type FetchCalendarProviderOptions = {
+  maxExpiration?: number;
+  active?: boolean;
+  resourceId?: string;
+};
+
+export interface CalendarDataProvider<
+  CalendarDataObject = DefaultCalendarDataObject,
```

Comment:
so the only thing I would maybe suggest is that here we should not only assign default but also extend the some BaseCalendarDataObject type. This will mean that our interface will then return or ask for an object that at minimum has the properties from the extended type - otherwise we could provide any value for the generic and implement this interface anywhere. That is btw also an option since all of our data-providers only have the 4 methods with similar arguments and returns but I don't think thats what we are aiming for here

Reply:
I was wondering about this too. The data providers and the controllers have some standard methods on them.

I left it for a refactoring task later on.

Reply:
Shall we try pairing on it together to see how it looks?